crypto/rand/prov_seed.c

```diff
@@ -38,14 +40,46 @@ size_t ossl_rand_get_entropy(ossl_unused const OSSL_CORE_HANDLE *handle,
     return ret;
 }
 
-void ossl_rand_cleanup_entropy(ossl_unused const OSSL_CORE_HANDLE *handle,
+size_t ossl_rand_get_user_entropy(OSSL_LIB_CTX *ctx,
+                                  unsigned char **pout, int entropy,
+                                  size_t min_len, size_t max_len)
+{
+    unsigned char *buf;
+    EVP_RAND_CTX *rng = ossl_rand_get0_seed_noncreating(ctx);
+    size_t ret;
+
+    if (rng == NULL)
+        return ossl_rand_get_entropy(ctx, pout, entropy, min_len, max_len);
+
+    /* Determine how many bytes to generate */
+    ret = entropy > 0 ? (size_t)(7 + entropy) / 8 : min_len;
+    if (ret < min_len)
+        ret = min_len;
+    else if (ret > max_len)
+        ret = max_len;
+
+    /* Allocate the return buffer */
+    if ((buf = OPENSSL_secure_malloc(ret)) == NULL)
+        return 0;
+
+    /* Fill the buffer */
+    if (!EVP_RAND_generate(rng, buf, ret, entropy, 0, NULL, 0)) {
+        OPENSSL_free(buf);
```

Comment:
OPENSSL_secure_clear_free ?

Reply:
I could but it seems pointless if the generate call fails.  It's not leaking anything IMO and the RNG will not operate again afterwards.

Could that be reverse engineered back to the seed material/DRBG state?  I doubt it.

Reply:
Agreed. The other existing functions do this.. Maybe they shouldnt at some point (it should really be the caller the thing called that does a clear if it needed to)